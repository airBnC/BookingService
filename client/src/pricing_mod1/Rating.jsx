import React from 'react';
import PropTypes from 'prop-types';
import ReactStars from 'react-stars';
import { abnbDarkBlue, Holder } from '../ModStylings';

export default function Rating(props) {
  if (props.rating) {
    return (
      <Holder mod1>
        <ReactStars
          count={5}
          value={props.rating}
          color2={abnbDarkBlue}
          edit={false}
        />
        <div style={{ marginLeft: '7px' }}>
          {props.numReviews}
        </div>
      </Holder>
    );
  }
  return null;
}

Rating.propTypes = {
  rating: PropTypes.number,
  numReviews: PropTypes.number,
};
Rating.defaultProps = {
  rating: 0,
  numReviews: 0,
};
