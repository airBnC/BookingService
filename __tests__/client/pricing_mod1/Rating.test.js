import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import 'jest-styled-components';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ReactStars from 'react-stars';
import Rating, { Holder, NumReviews } from '../../../client/src/pricing_mod1/Rating';

configure( { adapter: new Adapter() } );

const props = {
  rating: 0,
  numReviews: Math.floor( Math.random() * 2000 ),
  ratingNull: undefined,
};

describe( 'Holder Component', () => {
  it( 'is the main flex component', () => {
    const tree = renderer
      .create( <Holder> things inside </Holder> )
      .toJSON();
    expect( tree ).toHaveStyleRule( 'display', 'flex' );
    expect( tree ).toMatchSnapshot();
  } );
} );
describe( 'Rating Component', () => {
  it( 'renders number of reviews correctly', () => {
    const tree = renderer
      .create( <NumReviews> {props.numReviews}</NumReviews> )
      .toJSON();
    const expected = [ ' ', `${ props.numReviews }` ];
    expect( tree.children )
      .toEqual( expect.arrayContaining( expected ) );
  } );
  it( 'does not render if rating is undefined', () => {
    const tree = renderer
      .create( <Rating rating={props.ratingNull} /> )
      .toJSON();
    expect( tree ).toMatchSnapshot();
  } );
} );
