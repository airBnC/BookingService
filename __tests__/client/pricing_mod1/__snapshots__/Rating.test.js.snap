// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component holder contains both price per night and number of reviews 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: inherit;
  padding-top: 10px;
}

<div
  className="c0"
>
   things inside 
</div>
`;

exports[`Rating component reviews does not render if rating is undefined 1`] = `null`;
